lib/common/grpc-service.js

```diff
@@ -325,7 +325,8 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
 
     request: function() {
       return service[protoOpts.method](reqOpts, grpcOpts)
-        .on('status', function(status) {
+        .on('metadata', function() {
+          var status = this.received_status || { code: 0 };
```

Comment:
This is not a reliable way to get the status, AFAIK two requests are made - one sets the status the other emits the metadata. Per https://github.com/GoogleCloudPlatform/gcloud-node/issues/1443#issuecomment-234402654 the `metadata` event seems to be the way to go regardless. I can remove the check for `received_status` if you prefer.